lib/sequel/plugins/enum_values.rb

```diff
@@ -31,18 +34,18 @@ def self.apply(model, _options = {})
 			##   Item.plugin :enum_values, predicate_methods: %[status type]
 			## @example Define predicate methods for specific enum field
 			##   Item.plugin :enum_values, predicate_methods: :status
+			## @example Define filter methods for all enum fields
+			##   Item.plugin :enum_values, filter_methods: true
+			## @example Define filter methods for specific enum field
+			##   Item.plugin :enum_values, filter_methods: :status
+			## @example Define filter methods for specific enum fields
+			##   Item.plugin :enum_values, filter_methods: %[status type]
 			def self.configure(model, options = {})
 				model.instance_exec do
+					@enum_values_options = options
 					@enum_values_caching = options.fetch(:caching, @enum_values_caching)
 
-					predicate_methods = options.fetch(:predicate_methods, false)
-
-					transform_predicate_methods_to_enum_fields(predicate_methods)
-						.each do |field|
-							all_enum_fields[field][:enum_values].each do |enum_value|
-								define_predicate_method field, enum_value
-							end
-						end
+					process_enum_values_options
```

Comment:
1. If you're saving local variable `options` into instance variable `@enum_values_options` — please, work with it later, like for `@enum_values_caching`.
2. If you have a method like `process_enum_values_options` — you should place into it all operations with `options`, like `@enum_values_caching = options.fetch`.
3. Maybe the whole content of `process_enum_values_options` method can fit here, in the `self.configure` method, but I'm not sure.

Reply:
moved cahing into `process_enum_values_options`.

i was unable to put the content of `process_enum_values_options` in `self.configure`